iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -452,33 +452,46 @@ public override Task RejectAsync(string lockToken, CancellationToken cancellatio
 
         protected override void Dispose(bool disposing)
         {
-            if (_disposed)
-            {
-                return;
-            }
-
-            base.Dispose(disposing);
```

Comment:
removed the call to `base.Dispose()`. The transport handler Dispose calls are being made through the delegating handler "inner handler" dispose call. So calling back the base class dispose isn't required.
https://github.com/Azure/azure-iot-sdk-csharp/blob/bf5b6997486fa47691efd5fba999db2f2078b7f8/iothub/device/src/Pipeline/DefaultDelegatingHandler.cs#L229

Reply:
I've reverted this change for now. I'll rework the inheritance disposal pattern of our pipeline (if necessary) in the next PR. I'm running some tests to confirm the behavior.